onnxruntime/core/providers/cpu/controlflow/loop.cc

```diff
@@ -411,6 +438,7 @@ void LoopImpl::SaveOutputsAndUpdateFeeds(const std::vector<OrtValue>& last_outpu
 
   // save loop outputs as we have to concatenate at the end
   for (int j = info_.num_loop_carried_vars; j < info_.num_outputs; ++j) {
+    ORT_ENFORCE(last_outputs[j + 1].IsTensor(), "All scan outputs MUST be tensors");
     loop_output_tensors_[j - info_.num_loop_carried_vars].push_back(last_outputs[j + 1]);  // skip 'cond' in output
```

Comment:
This is enforced by the onnx type constraint checking isn't it?

Reply:
Is there something in the framework that ensures that the type of the OrtValue will be the same as the ONNX inferred type ?